pkg/workflows/create_test.go

```diff
@@ -310,3 +316,48 @@ func TestCreateWorkloadClusterRunSuccess(t *testing.T) {
 		t.Fatalf("Create.Run() err = %v, want err = nil", err)
 	}
 }
+
+func TestCreateWorkloadClusterTaskFailure(t *testing.T) {
+	test := newCreateTest(t)
+	commandContext := task.CommandContext{
+		BootstrapCluster: test.bootstrapCluster,
+		ClusterSpec:      test.clusterSpec,
+		Provider:         test.provider,
+		ClusterManager:   test.clusterManager,
+	}
+
+	gomock.InOrder(
+		test.clusterManager.EXPECT().CreateWorkloadCluster(
+			test.ctx, test.bootstrapCluster, test.clusterSpec, test.provider,
+		).Return(nil, errors.New("test")),
+		test.clusterManager.EXPECT().SaveLogsManagementCluster(
+			test.ctx, test.clusterSpec, test.bootstrapCluster,
+		),
+		test.clusterManager.EXPECT().SaveLogsWorkloadCluster(
```

Comment:
This one should not be executed, right? It should only be executed if `CreateWorkloadCluster` succeeds (if not, there is not kubeconfig)

Reply:
Currently how it is written, it get's called but then returns early. We can add a condition to not call this method instead but wanted to keep it as it is.

Reply:
That's why the last arg is nil